src/app/tasks/task-list/task-list.component.ts

```diff
@@ -14,11 +15,17 @@ import { BasicTask } from '../task.model';
   templateUrl: './task-list.component.html',
   styleUrl: './task-list.component.scss'
 })
-export class TaskListComponent {
+export class TaskListComponent implements OnDestroy {
 
-  @Input() taskList: BasicTask[] = this.taskService.getTasks();
+  @Input() taskList: BasicTask[] = this.taskService.tasksSubject.getValue();
+  private taskListSubscription : Subscription = this.taskService.tasksSubject.subscribe(
```

Comment:
you should prioritize using observables instead of subscribing to the results each time a request is made, that way you can not only improve performance but also code readability (you can subscribe automatically to observables in the template by using async pipe)